packages/frontend/src/app/Event/page.tsx

```diff
@@ -0,0 +1,54 @@
+'use client';
+
+import React, { useEffect, useState } from 'react';
+
+import EventsDisplay from '@/components/molecules/EventsDisplay';
+import EventSearchPanel from '@/components/molecules/EventSearchPanel';
+
+interface Event {
+  id: number;
+  date: string;
+  description: string;
+  flyer_link: string;
+}
+
+const EventPage = () => {
+  const [events, setEvents] = useState<Event[]>([]);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const fetchEvents = async () => {
+      try {
+        const response = await fetch('/_data/MOCK_DATA.json');
```

Comment:
pages should do the fetching themselves
we need to centralize the API fetching logic in one place


Refer to this link: https://semaphoreci.com/blog/next-js-site-api-layer